lib/paxmex/schema/field.rb

```diff
@@ -1,59 +1,69 @@
 require 'bigdecimal'
 require 'paxmex/schema'
 
-class Paxmex::Schema::Field
-  DATE_PATTERN = /^date\((.+)\)$/
-  TIME_PATTERN = /^time\((.+)\)$/
+module Paxmex
+  class Schema
+    class Field
+      DATE_PATTERN = /^date\((.+)\)$/
+      TIME_PATTERN = /^time\((.+)\)$/
 
-  attr_reader :name, :start, :final, :type
+      attr_reader :name, :start, :final, :type
 
-  def initialize(opts = {})
-    @name  = opts[:name]
-    @start = opts[:start]
-    @final = opts[:final]
-    @type  = opts[:type] || 'string'
-  end
+      def initialize(opts = {})
+        @name  = opts[:name]
+        @start = opts[:start]
+        @final = opts[:final]
+        @type  = opts[:type] || 'string'
+      end
 
-  def parse(raw_value)
-    case type
-    when 'string' then raw_value.rstrip
-    when 'julian' then parse_julian_date(raw_value) rescue nil
-    when 'date' then Date.strptime(raw_value, '%m%d%Y') rescue nil
-    when 'numeric' then raw_value.strip.to_i
-    when 'decimal' then parse_decimal(raw_value)
-    when DATE_PATTERN then parse_date_pattern(raw_value) rescue nil
-    when TIME_PATTERN then parse_time_pattern(raw_value) rescue nil
-    else fail "Could not parse field type #{type}. Supported types: string, julian, date, numeric, decimal, date(format), time(format)"
-    end
-  end
+      def parse(raw_value)
+        case type
+        when 'string' then raw_value.rstrip
+        when 'julian' then parse_julian_date(raw_value) rescue nil
+        when 'date' then Date.strptime(raw_value, '%m%d%Y') rescue nil
+        when 'numeric' then parse_numeric(raw_value)
+        when 'decimal' then parse_decimal(raw_value)
+        when DATE_PATTERN then parse_date_pattern(raw_value) rescue nil
+        when TIME_PATTERN then parse_time_pattern(raw_value) rescue nil
+        else fail "Could not parse field type #{type}. Supported types: string, julian, date, numeric, decimal, date(format), time(format)"
+        end
+      end
 
-  private
+      private
 
-  def parse_date_pattern(value)
-    Date.strptime(value, DATE_PATTERN.match(type).captures.first)
-  end
+      def parse_date_pattern(value)
+        Date.strptime(value, DATE_PATTERN.match(type).captures.first)
+      end
 
-  def parse_time_pattern(value)
-    Time.strptime(value, TIME_PATTERN.match(type).captures.first)
-  end
+      def parse_time_pattern(value)
+        Time.strptime(value, TIME_PATTERN.match(type).captures.first)
+      end
 
-  def parse_decimal(value)
-    # fields _may_ end with a letter
-    unless value.match(/^[0-9]*[0-9A-R{}]$/)
-      fail "Unexpected value '#{value}' for field '#{name}'"
-    end
+      def parse_numeric(value)
+        value.strip!
+        return value.to_f if value.include?('.')
```

Comment:
Is this guaranteed to be the decimal delimiter? You could also use the `Float()` and `Integer()` kernel methods, which would work on a greater variety of inputs